File: leetcode/golang/LCR_128_xuan_zhuan_shu_zu_de_zui_xiao_shu_zi_lcof.go
func stockManagement(stock []int) int {
    i, j := 0, len(stock) - 1
    for i <= j {
        m := i + (j-i) / 2
        if stock[m] < stock[j] {
            j = m
        } else if stock[m] > stock[j] {
            i = m + 1
        } else {
            j--
        }
    }
    return stock[i]
}

File: leetcode/c++/LCR_128_xuan_zhuan_shu_zu_de_zui_xiao_shu_zi_lcof.cpp
// https://leetcode.cn/problems/xuan-zhuan-shu-zu-de-zui-xiao-shu-zi-lcof/description

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int stockManagement(vector<int>& stock) {
        int i = 0, j = stock.size() - 1;
        while (i <= j) {
            int m = i + (j - i) / 2;
            if (stock[m] > stock[j])
                i = m + 1;
            else if (stock[m] < stock[j])
                j = m;
            else 
                j = j - 1;
        }
        return stock[i];
    }
};

// 二分查找，充分利用数组之前为有序这个条件，时间复杂度为O(logn)，空间复杂度为O(1)